frontend/src/components/Product.js

```diff
@@ -17,7 +17,14 @@ export default function Product(props) {
           rating={product.rating}
           numReviews={product.numReviews}
         ></Rating>
-        <div className="price">${product.price}</div>
+        <div className="row">
+          <div className="price">${product.price}</div>
+          <div>
+            <Link to={`/seller/${product.seller._id}`}>
+              {product.seller.seller.name}
```

Comment:
To me only works if I use {product.seller.name}

Reply:
also happened to me

Reply:
well actually name will not display with {product.seller.name}, but original code wont run neither

Reply:
any idea how to solve it?

Reply:
If you are stuck in this part maybe you need to delete mongodb database and recreate it, that solved my problem.

Reply:
As Jorge said, it solved it to me also.

Reply:
hello Jorge - how did you recreate it? you mean recreate the codes from scratch?

Reply:
I am too having trouble with this error. I deleted the whole database. However recreated the users using mongodb, its runs. But as soon as I add products the seller cannot be found.

Reply:
For those that are still wondering how to fix this issue, go to ProductRouters and check the /seed GET request. You will notice that each of the products are now mapped with the sellerId of a found seller in this case it is admin. 

so update your GET /seed route with that code , and then drop the database and then recreate it with the seed, you shouldn't have problems after.

Reply:
Hi Mj, I agree with Potato. I assigned all the products in mongodb with the admin seller id only and it works.

Reply:
All of my users have been extended Admin privileges since we learned how to edit users.  anvoluther's solution of assigning all products in mongodb with a seller id of user that is also an admin works for me, but only with my initial admin, not the other users who were recently changed to admin.

I am unsure why a seller must be an admin for it to work anyway.

Reply:
in productModel.js and orderModel.js have ObjectID, is necessary remplace to ObjectId. but in data.js verify if you have this:
users: [    {     
name: 'Basir',      
email: 'admin@example.com',      
password: bcrypt.hashSync('1234', 8),      
isAdmin: true,      
isSeller: true,      
**seller: {        
name: 'Puma',        
logo: '/images/logo1.png',        
description: 'best seller',        
rating: 4.5,        
numReviews: 120,      
},**   
 },

Reply:
> in productModel.js and orderModel.js have ObjectID, is necessary remplace to ObjectId. but in data.js verify if you have this:
> users: [ {
> name: 'Basir',
> email: '[admin@example.com](mailto:admin@example.com)',
> password: bcrypt.hashSync('1234', 8),
> isAdmin: true,
> isSeller: true,
> /* adding this part "seller" */
> **seller: { name: 'Puma', logo: '/images/logo1.png', description: 'best seller', rating: 4.5, numReviews: 120, },**
> 
> },
> you need delete the user's and post http://localhost:5000/api/users/seed

Thanks very much fher333. I just happened to revisit this project and encountered the same issue again lol but this time your recommendation fixed the issue.

Reply:
fher333 and anvoluther could you let us your code for the solution you propose, please ??

Reply:
TypeError: Cannot read property '_id' of null
Product
C:/Users/Sergio DC/Desktop/amazona/frontend/src/components/Product.js:27
  24 | <div className="row">
  25 |     <div className="price">${product.price}</div>
  26 |     <div>
> 27 |         <Link to={`/seller/${product.seller._id}`}>    
     | ^  28 |             {product.seller.seller.name}
  29 |         </Link>
  30 |     </div>

what happend with {product.seller._id} ??

Reply:
Hi Royal, you must add a seller for each product and also update the seller's name to get rid of the error. I have the same codes as yours but I did add a seller for all products and updated the seller name as well in User Profile.

Reply:
Thank you so much Anvoluther, It is a pleasure to have comrades as good as you

Reply:
At this point of the course, anyone more has the issue that normal user only can buy admin's products ??? 
It's happening to me, and I think I have the same code that Basir and you, but I am checking the past changes in this lessons' section and I don't know where is the issue...
please, could anyone test it?? just login as an new normal user and try to place order with products of an user that you made seller... and tell me if you can pleasee
PD: when I go to place order with a normal user to buy products of other normal seller user, the consoles says me that the issue is in the backend I think ...
POST http://localhost:3000/api/orders 500 (Internal Server Error)
any idea ?? 
you checked if you can do it ??

Reply:
@anvoluther @Royal6969 I have the same error: "TypeError: Cannot read property '_id' of undefined". I tried following what you said but I do not understand. Can you please explain what changes I needed to make exactly in productModel.js and orderModel.js?

Reply:
Hi gannau, in product model and order model - seller is added with below line:
 seller: { type: mongoose.Schema.Types.ObjectId, ref: "User" },

It's hard to tell what went wrong on your part since I don't have your codes but this error started showing up with the change to multi-vendor but if you assigned each of the product with a seller id which in this case is Admin, there shouldn't be a problem. The other option is to delete old products and create a new one either in frontend or with /seed in backend but make sure to update the seller name and it doesn't have to be just admin this time.

Reply:
@gannau 
you have to delete all your products and create again IN THE PAGE throw admin view and seller view for example 
use delete and create product button that you had done before 

In my case, one day, doing this lesson, 
I found my users deleted and had to create it with login screen and admin view for privileges (seller or admin) and I checked the data in Moongose too

After that, I delete and create again all product
For the admin, his product first
and for sellers their products too 
for separate
both are new user , remember?? 

And later with the Basir's code, all began to work perfectly

PD: In some cases, ObjectId in MongoDB gave me issues and it got me crazy but with moongose view (compass) is easy manage the data but it's mmmmm ...
Also, check your ObjectId in sellers!! 
you have to be same ObjectId for group of product in common 
Also if you experiment the same issue you are showing me in past email, 
but instead with ._id , you have the same issue with name
it mean that you would had previously filled the fields for each seller in their seller views in the page

Reply:
> @gannau
> you have to delete all your products and create again IN THE PAGE throw admin view and seller view for example
> use delete and create product button that you had done before
> 
> In my case, one day, doing this lesson,
> I found my users deleted and had to create it with login screen and admin view for privileges (seller or admin) and I checked the data in Moongose too
> 
> After that, I delete and create again all product
> For the admin, his product first
> and for sellers their products too
> for separate
> both are new user , remember??
> 
> And later with the Basir's code, all began to work perfectly
> 
> PD: In some cases, ObjectId in MongoDB gave me issues and it got me crazy but with moongose view (compass) is easy manage the data but it's mmmmm ...
> Also, check your ObjectId in sellers!!
> you have to be same ObjectId for group of product in common
> Also if you experiment the same issue you are showing me in past email,
> but instead with ._id , you have the same issue with name
> it mean that you would had previously filled the fields for each seller in their seller views in the page

Reply:
I found that at models the ObjectId was written like this: ObjectID... and it should be ObjectId. When I changed that, it worked perfect :)